src/app/dog/dog.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { DogsService } from '../dogs/dogs.service';
+
+@Component({
+	selector: 'app-dog',
+	templateUrl: './dog.component.html',
+	styleUrls: ['./dog.component.css']
+})
+export class DogComponent implements OnInit {
+
+	public subBreeds: any[] = [];
+	public breed: string = "";
+	public imagePath: string = "";
+
+	constructor(
+		private readonly dogsService: DogsService,
+		private readonly activatedRoute: ActivatedRoute,
+		private readonly router: Router
+	) { }
+
+	public async ngOnInit() {
+		this.activatedRoute.params.subscribe(
+			(params: Params) => {
+				this.breed = params['name'];
+			}
+		);
```

Comment:
to be sure if the breed is got from URL params, move this to constructor scope
constructor() is executed before ngOnInit()